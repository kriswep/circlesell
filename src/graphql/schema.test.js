/* globals test expect */
import typeDefs from './schema';

test('schema definition should be the right one', () => {
  expect(typeDefs).toMatchSnapshot();
});
