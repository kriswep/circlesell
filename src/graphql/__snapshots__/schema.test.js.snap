// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema definition should be the right one 1`] = `
"schema {
  query: Query
  mutation: Mutation
}

type Query {
  manufacturers(offset: Int!, limit: Int!, id: Int, name: String): [Manufacturer]
  manufacturer(id: Int!): Manufacturer
  user: Author
  posts(offset: Int!, limit: Int!, id: Int, title: String, text: String): [Post]
  post(id: Int!): Post
  authors(offset: Int!, limit: Int!, id: Int, firstName: String, lastName: String): [Author]
  author(id: Int!): Author
  products(offset: Int!, limit: Int!, id: Int, name: String): [Product]
  product(id: Int!): Product
  quantitys(offset: Int!, limit: Int!, id: Int, name: String): [Quantity]
  quantity(id: Int!): Quantity
}

type Mutation {
  upvotePost(postId: Int!): Post
}

type Manufacturer {
  id: Int!
  name: String
  products: [Product]
}

type Author {
  id: Int!
  firstName: String
  lastName: String
  posts: [Post]
}

type Post {
  id: Int!
  title: String
  text: String
  author: Author
  votes: Int
}

type Product {
  id: Int!
  name: String
  description: String
  manufacturer: Manufacturer
  amount: Quantity
}

type Quantity {
  id: Int!
  amount: Int
  product: Product
}
"
`;
